addons/portal/wizard/portal_invite_share.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models, _
+
+
+class PortalInviteShare(models.TransientModel):
+    _name = 'portal.invite.share'
```

Comment:
I would prefer a better name. I suppose portal.share could be sufficient. As for guidelines, once updated you will have to update xml ids and view names so choose something very nice :game_die:  !